### PR TITLE
:sparkles: Add commands parsing

### LINKED ISSUE #8: :scissors: Commands Parsing
# 👮‍♂️ What's needed
We will need a feature to parse the commands the users send and to ensure that the command is valid. 

# 📜 Sources
image

### PR DESCRIPTION
**Pull request by Marius-P1**
I added the main logic and the command parsing for the client part.
I still need to add all the commands we are going to use to communicate with the server

### CODE REVIEW

File: client/src/args_handling/args_checking.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+*/
+
+#include "myclient.h"
+
+static bool is_valid_ip(char *ip)
+{
+    int dots = 0;
+    int digits = 0;
+
+    for (int i = 0; ip[i]; i++) {
```
Comment (axel-eck) [inline client/src/args_handling/args_checking.c]: `i` shall have been a `size_t`

File: client/src/client.c [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+int client(int ac, char **av)
+{
+    client_t *client = NULL;
+    host_t *host = malloc(sizeof(host_t));
+    api_handler_t *api_handler = NULL;
+
+    if (host == NULL) {
+        perror("malloc");
```
Comment (axel-eck) [inline client/src/client.c]: Printing in stderr is forbidden

File: client/src/commands/on_command.c [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/*
+** EPITECH PROJECT, 2024
+** MyTeams
+** File description:
+** on_command
+*/
+
+#include "myclient.h"
+
+// void on_command(char *cmd, client_t *client)
```
Comment (axel-eck) [inline client/src/commands/on_command.c]: Remove this

File: client/src/commands/parsing/parsing.c [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+** MyTeams
+** File description:
+** parsing
+*/
+
+#include "myclient.h"
+
+static void skip_spaces(char *cmd, int *index)
+{
+    for (; cmd[*index] != '\0' && cmd[*index] == ' '; (*index)++);
```
Comment (axel-eck) [inline client/src/commands/parsing/parsing.c]: Please use while loops for loops you don't know the number of iterations

File: client/src/utils/utils.c [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+** utils
+*/
+
+#include "myclient.h"
+
+int tab_len(char **tab)
+{
+    int i = 0;
+
+    for (; tab[i] != NULL; i++);
```
Comment (axel-eck) [inline client/src/utils/utils.c]: While loop and size_t please

**Review by axel-eck — CHANGES_REQUESTED** [review]

**Review by 6im0n — DISMISSED** [review]
Good for me !

**Review by axel-eck — CHANGES_REQUESTED** [review]
Please remove all perror

File: client/src/client.c [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+    status = read(fd, buf, 1);
+    if (status == -1)
+        exit(84);
+    if (status == 0) {
+        printf("Exiting client\n");
+        *running = false;
+        return false;
+    }
+    if (status != -1 && status != 0) {
```
Comment (axel-eck) [inline client/src/client.c]: status != 0 is always true here

**Review by 6im0n — DISMISSED** [review]

File: client/src/client.c [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+    while (running) {
+        if (client->waiting_for_response == false) {
+            shell(client, readfds, buffer, &running);
+        }
+        if (client->api_handler->running)
+            network_manager_handle_waiting_sockets(
+                client->api_handler->manager);
+    }
+    return 0;
```
Comment (axel-eck) [inline client/src/client.c:104]: this should not return anything since it's always 0

File: client/src/commands/help.c [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+static void compare_help(char **parsed_cmd)
+{
+    if (strcmp(parsed_cmd[1], "create") == 0)
+        help_create();
+    if (strcmp(parsed_cmd[1], "list") == 0)
+        help_list();
+    if (strcmp(parsed_cmd[1], "info") == 0)
+        help_info();
+    else
+        printf("The command %s either does not exist "
```
Comment (axel-eck) [inline client/src/commands/help.c]: This will be ran for any command that is not "info"

File: client/src/commands/on_command.c [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+#include "myclient.h"
+
+void on_command(char *cmd, client_t *client)
+{
+    char **parsed_cmd = parse_command(cmd);
+
+    if (parsed_cmd == NULL)
+        return;
+    if (strcmp(parsed_cmd[0], "/help") == 0)
+        return help(parsed_cmd);
```
Comment (axel-eck) [inline client/src/commands/on_command.c:17]: on_command returns void, please split those two lines

**Review by axel-eck — CHANGES_REQUESTED** [review]
Still some small issue

**Review by axel-eck — APPROVED** [review]

**Review by 6im0n — APPROVED** [review]
OK for me !

**Pull request merged**